### ISSUE TITLE
Applying YOLOE to X-ray Object Detection

### ISSUE THREAD

**Issue by adgai1** [ultralytics/ultralytics#22252]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, I’m working on applying YOLOE to X-ray image based object detection. I’d like to ask a few questions:

Do I need to retrain the model from scratch for the X-ray domain, or would fine-tuning on a pretrained YOLOE model be sufficient?

Roughly how large should the dataset be to achieve good performance in such a domain-specific scenario?

If I mix X-ray samples with the original datasets (e.g., COCO or Objects365), will the existing categories interfere with the detection of new X-ray categories? Or would it be better to train on a separate dataset for this domain?

Thanks a lot for your guidance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @adgai1, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
> Do I need to retrain the model from scratch for the X-ray domain, or would fine-tuning on a pretrained YOLOE model be sufficient?

You don't train from scratch unless you have tens of thousands of images

> Roughly how large should the dataset be to achieve good performance in such a domain-specific scenario?

There's no specific number. Probably a few thousand depending on the number of classes. More classes require more images.

> If I mix X-ray samples(e.g., PiXray) with RGB datasets (e.g., COCO or Objects365), will the existing RGB categories interfere with the detection of X-ray categories due to the domain gap between RGB and X-ray? Or would it be better to train on a separate X-ray dataset for this domain?

What's the reason for mixing?

**Comment by adgai1** [comment]
I’m currently working on X-ray object detection using YOLOE. I have around 30k training samples across 12 classes. I attempted to fine-tune the model with the following code:

from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEPETrainer
import torch

model = YOLOE("ultralytics/cfg/models/v8/yoloe-v8s.yaml")
data = "xray.yaml"

model.train(
    data=data,
    epochs=80,
    close_mosaic=10,
    optimizer="AdamW",
    lr0=1e-3,
    warmup_bias_lr=0.0,
    weight_decay=0.025,
    momentum=0.9,
    workers=4,
    device="0",
    trainer=YOLOEPETrainer,
    pretrained="yoloe-v8s-seg.pt",
)


However, the results are not satisfactory:

One or two classes in my dataset achieve very low performance.

<img width="3000" height="2250" alt="Image" src="https://github.com/user-attachments/assets/389234e9-6388-447f-a043-24fb00a59bff" />

Fine-tuning seems to cause the model to lose its open-vocabulary capabilities.

I’m considering whether I should mix my X-ray samples with the original datasets and train from scratch if my current sample size is insufficient. Also, I would like advice on how to specifically improve performance for these low-performing classes.

**Comment by Y-T-G** [comment]
> Fine-tuning seems to cause the model to lose its open-vocabulary capabilities.

Yes, it will become closed set. And even if it doesn't, fine-tuning would heavily degrade performance on other classes because of catastrophic forgetting. That's just inevitable.

> I’m considering whether I should mix my X-ray samples with the original datasets and train from scratch if my current sample size is insufficient. Also, I would like advice on how to specifically improve performance for these low-performing classes.

If you're planning to detect both RGB and X-ray, sure. But they are very different modalities, so it may not perform well

**Comment by adgai1** [comment]
> > Fine-tuning seems to cause the model to lose its open-vocabulary capabilities.
> 
> Yes, it will become closed set. And even if it doesn't, fine-tuning would heavily degrade performance on other classes because of catastrophic forgetting. That's just inevitable.
> 
> > I’m considering whether I should mix my X-ray samples with the original datasets and train from scratch if my current sample size is insufficient. Also, I would like advice on how to specifically improve performance for these low-performing classes.
> 
> If you're planning to detect both RGB and X-ray, sure. But they are very different modalities, so it may not perform well

Understood, I will try other way to achieve this.

**Comment by glenn-jocher** [comment]
Sounds good—if you’re X‑ray only, stick with linear‑probing fine‑tuning on YOLOE‑11 and disable color-heavy augs; use the detection graph initialized from the matching seg checkpoint (don’t pass `pretrained=...`, load weights explicitly), and ensure inputs are 3‑channel (repeat grayscale) to match model expectations:

```python
from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEPETrainer

model = YOLOE("yoloe-11s.yaml")        # detection config
model.load("yoloe-11s-seg.pt")         # init from seg checkpoint (same scale)

results = model.train(
    data="xray.yaml",
    epochs=80,
    trainer=YOLOEPETrainer,            # linear probing (freezes most layers)
    mosaic=0.0, mixup=0.0,             # reduce domain gap for X-ray
    hsv_h=0.0, hsv_s=0.0, hsv_v=0.0,
)
```

If you still need open‑vocab elsewhere, keep a separate untouched YOLOE for that use case; more details are in the [YOLOE fine‑tuning guide](https://docs.ultralytics.com/models/yoloe/).
